php/Terminus/Auth.php

```diff
@@ -20,59 +20,75 @@ public function __construct() {
   }
 
   /**
-   * Determines if user is logged in
+   * Ensures the user is logged in or errs.
    *
-   * @return [boolean] True if user is logged in
+   * @return [boolean] Always true
    */
-  public static function loggedIn() {
-    if (Session::instance()->getValue('session', false) === false) {
-      throw new TerminusException(
-        'Please login first with `terminus auth login`',
-        array(),
-        1
-      );
+  public static function ensureLogin() {
+    $session = Session::instance()->getData();
+    $auth    = new Auth();
+    if (!$auth->loggedIn()) {
+      if (isset($session->refresh)) {
+        $auth->logInViaRefreshToken($session->refresh);
+      } else {
+        throw new TerminusException(
+          'Please login first with `terminus auth login`',
+          array(),
+          1
+        );
+      }
     }
     return true;
   }
 
   /**
+   * Checks to see if the current user is logged in
+   *
+   * @return [boolean] $is_logged_in True if the user is logged in
+   */
+  public function loggedIn() {
+    $session = Session::instance()->getData();
+    $is_logged_in = (
+      isset($session->session)
+      && (Terminus::isTest() || ($session->session_expire_time >= time()))
+    );
+    return $is_logged_in;
+  }
+
+  /**
    * Execute the login based on an existing session token
    *
    * @param [string] $token Session token to initiate login with
    * @return [boolean] True if login succeeded
    */
-  public function logInViaSessionToken($token) {
+  public function logInViaRefreshToken($token) {
     $options = array(
       'headers' => array('Content-type' => 'application/json'),
-      'cookies' => array('X-Pantheon-Session' => $token),
+      'body'    => array(
+        'refresh_token' => $token,
+      ),
     );
 
-    $this->logger->info('Validating session token');
-    $response = TerminusCommand::request('user', '', '', 'GET', $options);
+    $this->logger->info('Logging in via refresh token');
+    $response = TerminusCommand::request('auth/refresh', '', '', 'POST', $options);
 
     if (!$response
       || !isset($response['info']['http_code'])
       || $response['info']['http_code'] != '200'
     ) {
       throw new TerminusException(
-        'The session token {token} is not valid.',
-        array('token' => $token),
+        'The refresh token {token} is not valid.',
+        compact('token'),
```

Comment:
Right, but since a refresh token is as good as a password if it gets stolen or leaked, my instinct is we should treat it as one and scrub it from logs and output